File: Codeforces/C++/Round #731 (Div. 3)/F.cpp
/*-----------------------------------------------------------*/
/* URL    : https://codeforces.com/contest/1547/problem/F
/* Title  : Array Stabilization (GCD version)
/* Notes  : tag-codeforces, tag-problem-f, tag-div-3
/*-----------------------------------------------------------*/
/*-------- valentk777 --------*/
#include <bits/stdc++.h>
/*----------------------------*/
using namespace std;
/*----------------------------*/
#define fast_cin() ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL)
#define debug(...) fprintf(stderr, __VA_ARGS__), fflush(stderr)
#define time__(d) for (auto blockTime = make_pair(chrono::high_resolution_clock::now(), true); blockTime.second; debug("%s: %d ms\n", d, (int)chrono::duration_cast<chrono::milliseconds>(chrono::high_resolution_clock::now() - blockTime.first).count()), blockTime.second = false)
/*----------------------------*/
#define ll long long int
typedef long double ld;
/*----------------------------*/
#define forn(i, e) for (ll i = 0; i < e; i++)
#define forsn(i, s, e) for (ll i = s; i < e; i++)
#define rforn(i, s) for (ll i = s; i >= 0; i--)
#define rforsn(i, s, e) for (ll i = s; i >= e; i--)
#define max_arr(x, n) *max_element(x, x + n)
#define min_arr(x, n) *min_element(x, x + n)
#define max_vec(x) max_element(x.begin(), x.end())
#define min_vec(x) min_element(x.begin(), x.end())

#define test() ll t; cin >> t; while (t--)
#define dbg(x) cout<<#x<<" = "<<x<<endl;

/*----------------------------*/

void IO() {
    #ifndef ONLINE_JUDGE
    freopen("../input.txt", "r", stdin);
    freopen("../output.txt", "w", stdout);
    #endif
    fast_cin();
}

ll gcd(ll a, ll b) {
    return b == 0 ? a : gcd(b, a % b);
}

bool check_if_all_equal(const ll a[], ll n) {
    while (--n > 0 && a[n] == a[0])
        ;
    return n == 0;
}

void solve_to_slow() {
    ll n;
    cin >> n;

    ll a[n];

    forn(i, n) cin >> a[i];

    ll number = 0;
    bool all_equal = check_if_all_equal(a, n);

    while (!all_equal) {
        number++;

        ll first = a[0];
        all_equal = true;

        forn(i, n) {
            if (i + 1 != n) {
                a[i] = gcd(a[i], a[i + 1]);
            } else {
                a[i] = gcd(a[i], first);
            }

            if (i > 0 && all_equal && a[i] != a[i - 1]) {
                all_equal = false;
            }
        }
    }

    cout << number;
}

int main() {
    IO();

    test() {
#ifdef ONLINE_JUDGE
        solve_to_slow();
        #else
        time__("Run duration: ") solve_to_slow();
        #endif

        cout << endl;
    }

    return 0;
}

File: Codeforces/Python/Round #820 (Div. 3)/F.py
# -----------------------------------------------------------
# URL    : https://codeforces.com/contest/1729/problem/F
# Title  : Title
# Notes  : tag-codeforces, tag-problem-F, tag-div-3, tag-not-pass
# -----------------------------------------------------------


def solve():
    s = input()
    n = len(s)
    w, m = list(map(int, input().split()))  # length of substring, queries

    found = 0
    options = {_i: [] for _i in range(0, 9)}

    for _i in range(1, n - w + 2):
        if found == 9 * 2:
            break

        r = int(s[_i - 1:_i + w - 1]) % 9

        if len(options[r]) < 2:
            options[r].append(_i)
            found += 1

    for i in range(m):
        l, r, k = list(map(int, input().split()))
        vv = int(s[l - 1:r]) % 9

        results = set()

        for mod1 in range(0, 9):
            for mod2 in range(0, 9):
                if (mod1 * vv + mod2) % 9 == k and len(options[mod1]) > 0 and len(options[mod2]) > 0:
                    if mod1 == mod2 and len(options[mod1]) == 2:
                        results.add((options[mod1][0], options[mod1][1]))
                    else:
                        L1 = min(options[mod1])
                        L2 = min(options[mod2])

                        if L1 == L2:
                            continue

                        results.add((L1, L2))

        if len(results) > 0:
            results = sorted(results, key=lambda x: (x[0], x[1]))
            print(f"{results[0][0]} {results[0][1]}")
        else:
            print("-1 -1")


if __name__ == "__main__":
    t = int(input())

    for _ in range(t):
        solve()
